ansys_optical_automation/interop_process/Coating_converter_speos_zemax_Analysis.py

```diff
@@ -0,0 +1,398 @@
+
+        # Writing the file
+        file_id1.write('OPTIS - Coated surface file v1.0\n')
+        file_id1.write(name1 + "\n")
+        file_id1.write(str(nb_angle_of_incidence) + " " + str(nb_wavelength) + "\n")
+
+        # 1st line
+        for wavelength_index in range(nb_wavelength):
+            wavelength = round(wavelength_min + wavelength_index * (wavelength_max - wavelength_min) / (
+                    nb_wavelength - 1), 3)
+            file_id1.write("\t" + " " + myformat.format(speos_wavelength_units_um * wavelength) + " " + "\t")
+        file_id1.write("\n")
+        # Loop to read the data
+        for angle_index in range(nb_angle_of_incidence):
+            # 1st column: angle of incidence
+            angleofincidence = float(coating_data[angle_index][0])
+            file_id1.write(myformat.format(angleofincidence) + ' \t')
+            for wavelength_index in range(nb_wavelength):
+                offset = angle_index + wavelength_index * (nb_angle_of_incidence + 1)
+                reflectance_ppol_1 = float(coating_data[offset][2])
+                transmittance_ppol_1 = float(coating_data[offset][4])
+                file_id1.write(myformat.format(100 * reflectance_ppol_1) + '\t' + myformat.format(
+                    100 * transmittance_ppol_1) + '\t')
+            file_id1.write("\n\t")
+            for wavelength_index in range(nb_wavelength):
+                offset = angle_index + wavelength_index * (nb_angle_of_incidence + 1)
+                reflectance_spol_1 = float(coating_data[offset][1])
+                transmittance_spol_1 = float(coating_data[offset][3])
+                file_id1.write(myformat.format(100 * reflectance_spol_1) + "\t" + myformat.format(
+                    100 * transmittance_spol_1) + "\t")
+            file_id1.write("\n")
+        file_id1.close()
+        # print("File " + coatingfilename1 + " created")
+        coating_data.clear()
+        # os.remove(resultfullfilename)
+
+
+    def __init__(self,coatingfilename, coatingfolder,
+            substrate_catalog,substrate_name,
+            user_wavelength_min, user_wavelength_max, nb_wavelength, speos_wavelength_units_um,
+            nb_digits, skip_lines):
+        """
+            function that converts a Zemax coating / substrate into a speos coating file and a bsdf180
+            The function reads the data from the Transmission vs Angle analysis
+            The settings are not implemented in ZOS-API for this analysis.
+            So the settings by default should be set to
+            - incident angles  from 0 to 90
+            - surface to 1
+
+            Parameters
+            ----------
+            coatingfilename : string
+                Name of the coating file to convert
+            coatingfolder : string
+                Path to the coating file
+            substrate_catalog : string
+                Catalog of the materials
+            substrate_name : string
+                Names of the materials
```

Comment:
the input should be a list of names nor "a" name string

Reply:
I did the correction

Reply:
Let me know if that is ok. Thank you!